testing/sync-test/src/auth.rs

```diff
@@ -1,233 +1,46 @@
-            &self.cfg,
-            (&self.k_sync, &self.xcs),
-        )?;
-        let auth_params = auth::AuthorizationRequestParameters {
-            client_id: client_id.clone(),
-            code_challenge: Some(code_challenge.clone()),
-            code_challenge_method: Some(code_challenge_method.clone()),
-            scope: scope.clone(),
-            keys_jwe: Some(keys_jwe),
-            state: state.clone(),
-            access_type: "offline".to_string(),
-        };
-        auth::send_authorization_request(&self.cfg, auth_params, &auth_key)
-    }
-
-    fn execute_oauth_pair_flow(&self, oauth_uri: &str) -> Result<(String, String)> {
-        let url = Url::parse(oauth_uri)?;
-        let auth_params = auth::AuthorizationParameters::try_from(url)?;
-        let scoped_keys = auth::get_scoped_keys(
-            &auth_params.scope.join(" "),
-            &auth_params.client_id,
-            &auth::derive_auth_key_from_session_token(&self.session_token)?,
-            &self.cfg,
-            (&self.k_sync, &self.xcs),
-        )?;
-        // Setup authority account that is logged in and has the appropriate scoped keys
-        let fxa = FirefoxAccount::new_logged_in(self.cfg.clone(), &self.session_token, scoped_keys);
-
-        let state = auth_params.state.clone();
-        // Use the logged in client to generate the oauth code for
-        // a different client
-        let code = fxa.authorize_code_using_session_token(auth_params)?;
-        Ok((code, state))
+impl Account {
+    pub fn new(cfg: FxaConfig, cli: CliFxa) -> Self {
+        Self { cfg, cli }
     }
-}
 
-impl Drop for TestAccount {
-    fn drop(&mut self) {
-        if self.no_delete {
-            log::info!("Cleanup was explicitly disabled, not deleting account");
-            return;
-        }
-        log::info!("Cleaning up temporary firefox account");
-        let destroy_endpoint = self.cfg.auth_url_path("v1/account/destroy").unwrap();
-        let body = json!({
-            "email": self.email,
-            "authPW": auth::auth_pwd(&self.email, &self.pass).unwrap()
-        });
-        let req = Request::post(destroy_endpoint).json(&body).send();
-        match req {
-            Ok(resp) => {
-                if resp.is_success() {
-                    log::info!("Account destroyed successfully!");
-                    return;
-                } else {
-                    log::warn!("   Error: {}", resp.text());
-                }
-            }
-            Err(e) => log::warn!("   Error: {}", e),
-        }
-        log::warn!(
-            "Failed to destroy fxacct {} with pass {}!",
-            self.email,
-            self.pass
-        );
+    fn fxa_account(&self) -> &FirefoxAccount {
+        &self.cli.account
     }
 }
 
+// Many of these tests simulate multiple devices (aka clients).
 pub struct TestClient {
-    pub fxa: fxa_client::internal::FirefoxAccount,
-    pub test_acct: Arc<TestAccount>,
+    pub acct: Arc<Account>,
```

Comment:
What do you think about killing the `Account` struct all together and attaching its members here, looks like it only has one getter and it retrieved the inner `FirefoxAccount` so maybe we can just attach that?